maint/local_python/binding_c.py

```diff
@@ -187,6 +189,59 @@ def dump_proto_line(l, Out):
             dump_proto_line(re.sub(' MPI', ' PMPI', l, 1), Out)
         print("", file=Out)
 
+        # -- QMPI function enum --
+        print("enum QMPI_Functions_enum {", file=Out)
+        for func in G.FUNCS.values():
+            if 'dir' not in func:
+                continue
+            if re.match(r'MPI_DUP_FN', func['name']):
+                continue
+            print("    " + func['name'].upper() + "_T,", file=Out)
+        print("    MPI_LAST_FUNC_T", file=Out)
+        print("};", file=Out)
+        print("", file=Out)
+
+        # -- QMPI prototypes --
+        for l in G.mpi_declares:
+            if re.match(r'int MPI_DUP_FN', l):
+                continue
+            dump_proto_line(
+                    re.sub(', \)', ')',
+                        re.sub('\(', '(QMPI_Context context, int tool_id, ',
+                            re.sub(' MPI', ' QMPI',
+                                re.sub('\(void\)', '()',
+                                    l, 1),
+                                1),
+                            1),
+                        1),
+                    Out)
```

Comment:
> Enjoy your beautifully crafted art, but why not just:

Because then you wouldn't get my beautiful art. 😄 
I can do it either way. It doesn't matter.

Reply:
> On that, where does the last comma come from?

If the function has no arguments (i.e., `void`), then the extra comma comes from this line:

    re.sub('\(', '(QMPI_Context context, int tool_id, ',

I found it easier to remove it than to add an if statement to avoid adding it.

Reply:
I see. I would prefer an explicit `if` statement. It is "easier" when you write it, then you have to explain it to a later reviewer since the reviewer is too lazy to reason it out.

Reply:
How about we compromise and I add a comment...

Reply:
Okay, deal :)